### PR TITLE
Add support for internal data-groups.

### PR DESCRIPTION
**Pull request by f5gary**
Problem:
CCCL needs support for internal data-groups as they are required by
some iRules.

Solution:
- Updated the schema to understand internal data-groups.
- Added CRUD operations for the new data type.
- Updated existing unit tests to exercise the new data type.
- Added new unit tests for internal data-groups.

affects-branches: master

### CODE REVIEW

File: f5_cccl/service/manager.py [hunk 1]
```diff
@@ -269,13 +277,14 @@ def deploy(self, desired_config):  # pylint: disable=too-many-locals
 
         LOGGER.debug("Building task lists...")
         create_tasks = create_vaddrs + create_monitors + \
-            create_pools + create_irules + create_policies + \
-            create_virtuals + create_iapps
+            create_pools + create_irules + create_internal_data_groups + \
```
Comment (edarzins) [inline f5_cccl/service/manager.py]: I thought iRules can depend on internal_data_groups? If so, create the internal_data_groups before iRules.

File: f5_cccl/service/manager.py [hunk 2]
```diff
@@ -269,13 +277,14 @@ def deploy(self, desired_config):  # pylint: disable=too-many-locals
...
+            create_policies + create_virtuals + create_iapps
         update_tasks = update_vaddrs + update_monitors + \
-            update_pools + update_irules + update_policies + \
-            update_virtuals + update_iapps
+            update_pools + update_irules + update_internal_data_groups + \
```
Comment (edarzins) [inline f5_cccl/service/manager.py]: I thought iRules can depend on internal_data_groups? If so, update the internal_data_groups before iRules.

File: f5_cccl/service/manager.py [hunk 3]
```diff
@@ -269,13 +277,14 @@ def deploy(self, desired_config):  # pylint: disable=too-many-locals
...
+            update_policies + update_virtuals + update_iapps
         delete_tasks = delete_iapps + delete_virtuals + delete_policies + \
-            delete_irules + delete_pools + delete_monitors
+            delete_internal_data_groups + delete_irules + delete_pools + \
```
Comment (edarzins) [inline f5_cccl/service/manager.py]: I thought iRules can depend on internal_data_groups? If so, delete the internal_data_groups after iRules.

File: f5_cccl/testcommon.py [hunk 4]
```diff
@@ -360,6 +360,7 @@ def __init__(self):
         self.monitor = MockMonitor()
         self.virtuals = MockVirtuals()
         self.pools = MockPools()
+        self.data_group = MockPools()
```
Comment (edarzins) [inline f5_cccl/testcommon.py]: Looks like cut/paste mistake -- should be MockDataGroupInternals here?

**Comment by f5gary** [comment]
Agree with all comments and have addressed them with my 2nd commit.

**Pull request merged**